src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -185,15 +185,44 @@ module Make (Inputs : Inputs_intf.S) = struct
 
     let get_batch t locations =
       assert_is_attached t ;
-      let found_accounts, leftover_locations =
-        List.partition_map locations ~f:(fun location ->
-            match self_find_account t location with
-            | Some account ->
-                Either.first (location, Some account)
-            | None ->
-                Either.second location )
+      let locations_with_locations_rev, leftover_locations_rev =
```

Comment:
I suggest using the [version](https://github.com/MinaProtocol/mina/commit/8a5022369d5ae850cd882c3c681bd129eb387822)